cron/utils.ts

```diff
@@ -8,7 +16,54 @@ import config from 'config';
 export function onlyExecuteInProdOnMondays() {
   const today = new Date();
   if (config.env === 'production' && today.getDay() !== 1) {
-    console.log('OC_ENV is production and day is not Monday, script aborted!');
+    logger.warn('OC_ENV is production and day is not Monday, script aborted!');
     process.exit(0);
   }
 }
+
+export const runCronJob = async (
+  name: string,
+  run: () => Promise<any>,
+  timeoutMs: number,
+  errorParameters?: CaptureErrorParams,
+) => {
+  let exitCode = 0;
+  if (require.main !== module && process.env.DISABLE_CRON_MODULE_START_CHECK === undefined) {
+    logger.warn('This script is not meant to be required, please run it directly');
+    return;
+  }
```

Comment:
This env variable hacks around the problem but it will still exist in local dev. To me, the CRON job should never run when the file gets imported since you're not in control of when it will be executed. The right pattern for tests is to import the `main` function from tests and call it directly.

Or am I missing legit cases where we want the CRON job to start as soon as it's imported?

Reply:
The problem is the early return of this function. This is not a root call, it is an early return within `runCronJob`. 

How do we test a call to `runCronJob` if we necessarily need to import that as a module in the test file?

Reply:
@Betree I don't think the problem will still exist in local dev since DISABLE_CRON_MODULE_START_CHECK is never set at the environment level. It's just manipulated by the test case itself.

Reply:
Ok, I get it now. We would have handled this with a function parameter, but I'm fine with your approach:

```
export const runCronJob = async (
  name: string,
  run: () => Promise<any>,
  timeoutSeconds: number,
  { errorParameters, forceIfNotMain } = {}
) => {
```

Reply:
LOL. That was very obvious.

Reply:
LOL, `require.main !== module` is always true here, because we import this function from a file. 😂